daemon/algod/api/server/v1/handlers/handlers.go

```diff
@@ -442,6 +451,17 @@ func WaitForBlock(ctx lib.ReqContext, context echo.Context) {
 		}
 	}
 
+	nodeStatus, err := getNodeStatus(ctx.Node)
```

Comment:
If you implement this check as a middleware we can ensure that it is done for every endpoint, and we don't have to remember to add it to future endpoints.

Which endpoints that we want to allow even when the node is catchup up? At least the `DELETE` catchpoint call, what else?

Reply:
We clearly want this in any API that touches the transaction pool. 
I'm not sure how easy it would be to add this as a middleware. I would imagine that we need to export the spec tags into the code so that we can make these decision in runtime.

Reply:
We could decide to be overly aggressive and disable all<sub>1</sub> v1/v2 endpoints will be disabled during catchup. This may be a safer approach, since we wouldn't be able to forget the catchup check in future endpoints.

1: exceptions to all:
* `DELETE` + path.contains(`/v2/catchup/`)
* `GET` + path.contains(`v[\d]/status`)

Reply:
Isn't that being abit draconic ? I was trying to make sure that we limit the functionality only where it makes a difference. 
btw - there are few more exceptions in the common api.

Reply:
It might be, but I'm having trouble coming up with any legitimate uses for an out of date node.